scripts/irods/test/test_iquery.py

```diff
@@ -224,3 +225,66 @@ def test_genquery2_honors_group_permissions__issue_8259(self):
                     self.admin.run_icommand(['ichmod', '-M', '-r', 'own', self.user.username, self.user.session_collection])
                     self.admin.run_icommand(['iadmin', 'rmgroup', group])
                     self.admin.run_icommand(['iadmin', 'moduser', self.user.username, 'type', 'rodsuser'])
+
+    def test_genquery2_sql_generation_for_sql_distinct_keyword__issue_8261(self):
+        with self.subTest('GenQuery2 does not automatically insert DISTINCT keyword into SQL'):
+            _, out, _ = self.user.assert_icommand(['iquery', '--sql-only', 'select DATA_NAME'], 'STDOUT')
+            sql = out.strip()
+            self.assertNotIn('distinct', sql)
+            self.assertNotIn('select distinct', sql)
+
+        with self.subTest('GenQuery2 supports SELECT DISTINCT syntax'):
+            _, out, _ = self.user.assert_icommand(['iquery', '--sql-only', 'select distinct DATA_NAME'], 'STDOUT')
+            self.assertTrue(out.strip().startswith('select distinct '))
+
+        with self.subTest('GenQuery2 supports DISTINCT keyword in aggregate functions'):
```

Comment:
Is there much value in testing all of the aggregate functions? What about the order-by clause? Only asking because I noticed that order-by needed its own effort so I was wondering whether that and aggregate functions other than `count` had been covered.

Reply:
I don't think we need to cover specific aggregate functions since some things are likely to be RDMS dependent. What we can do is add a couple assertions which demonstrate the parser's forwarding ability of function names.

Will update test.

Reply:
Updated and verified changes pass.

Done.